diagram-viewer/src/main/java/com/powsybl/diagram/viewer/sld/SingleLineDiagramController.java

```diff
@@ -99,13 +111,19 @@ protected ContainerResult call() throws IOException {
                                     .setComponentLibrary(model.getComponentLibrary())
                                     .setSubstationLayoutFactory(model.getSubstationLayoutFactory())
                                     .setStyleProviderFactory(model::getStyleProvider)
-                                    .setVoltageLevelLayoutFactoryCreator(voltageLevelLayoutFactoryCreator);
-
-                            SingleLineDiagram.draw(network, container.getId(),
-                                    svgWriter,
-                                    metadataWriter,
-                                    sldParameters);
-
+                                    .setVoltageLevelLayoutFactoryCreator(voltageLevelLayoutFactoryCreator)
+                                    .setZoneLayoutFactory(model.getZoneLayoutFactory());
+                            if (container instanceof Network network) {
+                                SingleLineDiagram.drawMultiSubstations(network, ((Network) container).getSubstationStream().map(Identifiable::getId).toList(),
+                                        svgWriter,
+                                        metadataWriter,
+                                        sldParameters);
```

Comment:
I think we shouldn't do that. Every time a network is loaded then clicking on the full network will launch this heavy process which cannot be interrupted, together with the same kind of heavy process on network area diagram. You're probably forced to kill the whole viewer each time you click on this with a big network.

Plus I noticed that when the viewer isn't used for a long time, the tree view forgets the selection for some reason and then the full network is selected by default. It's already taking off my laptop without this, so I can't imagine what would occur with this extra computation :rocket: